src/api-service/__app__/onefuzzlib/workers/nodes.py

```diff
@@ -49,21 +49,25 @@ def create(
         machine_id: UUID,
         scaleset_id: Optional[UUID],
         version: str,
+        new: bool = False,
     ) -> "Node":
         node = cls(
             pool_name=pool_name,
             machine_id=machine_id,
             scaleset_id=scaleset_id,
             version=version,
         )
-        node.save()
-        send_event(
-            EventNodeCreated(
-                machine_id=node.machine_id,
-                scaleset_id=node.scaleset_id,
-                pool_name=node.pool_name,
+        # `save` only returns None when `new` is True and an object already
+        # exists in Azure Tables. If we're in that case, don't send an event.
+        result = node.save(new=new)
+        if result is None:
```

Comment:
Per the comment, did you want:
```suggestion
        if result is not None:
```

Reply:
The comment was worded oddly which lead to the misinterpretation.   Then I clicked the wrong button in github, which accepted this suggestion.  I reverted the suggestion and updated the comment.